l3kernel/l3doc.dtx

```diff
@@ -959,8 +959,8 @@ and all files in that bundle must be distributed together.
 %    \begin{macrocode}
 \bool_new:N \g_@@_typeset_documentation_bool
 \bool_new:N \g_@@_typeset_implementation_bool
-\bool_set_true:N \g_@@_typeset_documentation_bool
-\bool_set_true:N \g_@@_typeset_implementation_bool
+\bool_gset_true:N \g_@@_typeset_documentation_bool
+\bool_gset_true:N \g_@@_typeset_implementation_bool
```

Comment:
So there's no test file loading `l3doc` with `\debug_on:n { check-declarations }`.

Reply:
Did a test locally and caught one more: `\filesep` was set both globally and locally.
https://github.com/latex3/latex3/blob/7e5c76291cedd0d8c19f53e2c5cbafe9ab840553/l3kernel/l3doc.dtx#L3851
https://github.com/latex3/latex3/blob/7e5c76291cedd0d8c19f53e2c5cbafe9ab840553/l3kernel/l3doc.dtx#L3881

Reply:
Another: `\g_docinput_clist` was changed locally (line 3726).
https://github.com/latex3/latex3/blob/7e5c76291cedd0d8c19f53e2c5cbafe9ab840553/l3kernel/l3doc.dtx#L3722-L3729

Should I open a separate PR?

Reply:
Opened #1596.